osgp/platform/osgp-adapter-ws-admin/src/main/java/org/opensmartgridplatform/adapter/ws/admin/application/services/DeviceLogItemSpecifications.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.ws.admin.application.services;
+
+import static org.opensmartgridplatform.shared.utils.SearchUtil.replaceWildcards;
+
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensmartgridplatform.logging.domain.entities.DeviceLogItem;
+import org.springframework.data.jpa.domain.Specification;
+
+public final class DeviceLogItemSpecifications {
```

Comment:
org.opensmartgridplatform.domain.core.specifications contains interfaces for specifications. Any reason why this class does not implement such an interface?

Reply:
Made it in line with the other specification interfaces/classes. I'm not sure about the package for the implementation. I currently have:
Interface DeviceLogItemSpecifications: org.opensmartgridplatform.adapter.ws.admin.application.specifications
Class JpaDeviceLogItemSpecifications: org.opensmartgridplatform.adapter.ws.admin.infra.specifications

Reply:
OK